pandas/core/series.py

```diff
@@ -238,6 +239,11 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                 data = _sanitize_array(data, index, dtype, copy,
                                        raise_cast_failure=True)
 
+                if index is not None and len(index) != len(data):
```

Comment:
this should go a touch higher, 

```
            if index is None:
                if not is_list_like(data):
                    data = [data]
                index = com._default_index(len(data))
           else:
                # add here

            # create/copy the manager
            if isinstance(data, SingleBlockManager):
                if dtype is not None:
                    data = data.astype(dtype=dtype, errors='ignore',
                                       copy=copy)
```

Reply:
Ok. Added a scalar check that lets scalars through, so we are assuming that the Series is shaped correctly when the scalar is broadcast to fit the index, which is probably ok.
```
if index is None:
    ...
else:
    if isscalar(data) and len(index) != len(data):
        ...
```

Reply:
Nevermind. A few other inputs that break. `np.array` and `np.dtype` appear to be just two of them. Unless we add specific checks for these, I think we may need to move it lower, below `_sanitize_array`.